html/modules/custom/reliefweb_fields/src/Plugin/Field/FieldWidget/ReliefWebOptions.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Drupal\reliefweb_fields\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\OptionsButtonsWidget;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\OptionsWidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\reliefweb_utility\Traits\EntityDatabaseInfoTrait;
+
+/**
+ * Plugin implementation of the 'reliefweb_options' widget.
+ *
+ * @FieldWidget(
+ *   id = "reliefweb_options",
+ *   module = "reliefweb_fields",
+ *   label = @Translation("ReliefWeb Options with description"),
+ *   multiple_values = true,
+ *   field_types = {
+ *     "list_integer",
+ *     "list_string",
+ *     "list_float",
+ *     "boolean",
+ *     "entity_reference",
+ *   }
+ * )
+ */
+class ReliefWebOptions extends OptionsButtonsWidget {
+
+  use EntityDatabaseInfoTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $element = OptionsWidgetBase::formElement($items, $delta, $element, $form, $form_state);
+
+    $options = $this->getOptions($items->getEntity());
+    $selected = $this->getSelectedOptions($items);
+
+    // Load all terms, add description.
+    $entity_type_id = $this->getReferencedEntityTypeId();
+    $bundles = $this->getReferencedBundles();
+    $entity_type_manager = $this->getEntityTypeManager();
+
+    $entities = [];
+    foreach ($bundles as $bundle) {
+      if ($entity_type_id === 'taxonomy_term') {
+        $entities += $entity_type_manager->getStorage($entity_type_id)->loadByProperties([
+          'vid' => $bundle,
+        ]);
+      }
+      else {
+        $entities += $entity_type_manager->getStorage($entity_type_id)->loadByProperties([
+          'bundle' => $bundle,
```

Comment:
Media or nodes uses `type` not `bundle` for example. Also, as there is only one `entity_type_id`, maybe the whole foreach block can be simplified:


```php
$entities = $entity_type_manager->getStorage($entity_type_id)->loadByProperties([
  $this->getEntityTypeBundleField($entity_type_id) => $bundles,
]);
```